staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -104,12 +111,21 @@ type controller struct {
 // Controller is a low-level controller that is parameterized by a
 // Config and used in sharedIndexInformer.
 type Controller interface {
-	// Run does two things.  One is to construct and run a Reflector
+	// RunWithContext does two things.  One is to construct and run a Reflector
 	// to pump objects/notifications from the Config's ListerWatcher
 	// to the Config's Queue and possibly invoke the occasional Resync
 	// on that Queue.  The other is to repeatedly Pop from the Queue
 	// and process with the Config's ProcessFunc.  Both of these
-	// continue until `stopCh` is closed.
+	// continue until the context is canceled.
+	//
+	// It's an error to call RunWithContext more than once.
+	// RunWithContext blocks; call via go.
+	RunWithContext(ctx context.Context)
+
+	// Run does the same as RunWithContext with a stop channel instead of
+	// a context.
+	//
+	// TODO (https://github.com/kubernetes/kubernetes/issues/126379): logcheck:context // RunWithcontext should be used instead of Run in code which supports contextual logging.
```

Comment:
Still looks strange to not deprecate this (no blocker).

Reply:
I tried, and it would have a pretty big impact. Let's better not do that.